awswrangler/redshift.py

```diff
@@ -285,7 +291,10 @@ def _create_table(  # pylint: disable=too-many-locals,too-many-arguments
         if mode == "upsert":
             guid: str = uuid.uuid4().hex
             temp_table: str = f"temp_redshift_{guid}"
-            sql: str = f'CREATE TEMPORARY TABLE {temp_table} (LIKE "{schema}"."{table}")'
+            sql = f'CREATE TEMPORARY TABLE {temp_table} (LIKE "{schema}"."{table}")'
+            if columns:
+                col_names_str = ",".join(columns)
+                sql = f'CREATE TEMPORARY TABLE {temp_table} AS SELECT {col_names_str} FROM "{schema}"."{table}" LIMIT 0'
             _logger.debug(sql)
```

Comment:
@jaidisido @kukushking 
What do you think about CTAS with LIMIT 0 statement instead of CREATE ... LIKE statement?
Would it be correct to inherit a destintation/original table schema with appropriate column types and sizes for a staging?